tickets/SP-2211: Add rubin_sim.sim_archive.fetch_num_visits_in_latest_prenight_sim_for…

ehneilsen opened the pull request:
You can try it with:
```
import rubin_sim.sim_archive
rubin_sim.sim_archive.fetch_num_visits_in_latest_prenight_sim_for_night('2025-05-15')
```
or
```
import rubin_sim.sim_archive
rubin_sim.sim_archive.fetch_num_visits_in_latest_prenight_sim_for_night(20250515)
```

rhiannonlynne reviewed (changes requested):
See inline comment - a bit more ambitious about the API, but I think it'll last longer.

ehneilsen commented:
Okay, I've reworked the new function so we can enhance what we return later, and made the command name shorter.
Along the way to support this better, I refactored some of the functions it calls as well.
